beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -153,13 +153,15 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 			}
 
 			if err := s.validateBeaconBlock(ctx, b, blkRoot); err != nil {
-				log.Debugf("Could not validate block from slot %d: %v", b.Block().Slot(), err)
-				s.setBadBlock(ctx, blkRoot)
-				tracing.AnnotateError(span, err)
-				// In the next iteration of the queue, this block will be removed from
-				// the pending queue as it has been marked as a 'bad' block.
-				span.End()
-				continue
+				if !errors.Is(ErrOptimisticParent, err) {
```

Comment:
Will this evaluate first the child, find out that the parent is optimistic, continue without marking it as bad. Eventually when the EL syncs up, it seems to me that it will process only one block at a time, since it will start from the child.. eventually the loop reaches the top parent, this one is no longer optimistic and it passes validation. The next loop it will sync one block more. Shouldn't we sync blocks parent to child?